### PR TITLE
Adapt hello-world exercise metadata to be an Exercism introduction

### PR DESCRIPTION
**Pull request by ramonh**
This addresses exercism/rossconf#4, towards goal [#12](exercism/rossconf#12), by adapting the suggestion and text on exercism/exercism.io#2021. The purpose of this exercise is to serve as an introduction to Exercism.

The hello-world exercise will take an input for the caller's name. If a name is given, the program will say "Hello, #{name}!", and will otherwise spout "Hello, world!"

### CODE REVIEW

File: hello-world.md [hunk 1]
```diff
@@ -1,15 +1,30 @@
...
-Basically, a "Hello, World!" program simply outputs the text "Hello, World!" to the screen 
-or console. (NOTE: if you don't know what a console aka command-line interface aka command 
-window aka terminal window is, you should take a few minutes to 
-[read the Wikipedia article about the Command-line interface](http://en.wikipedia.org/wiki/Command-line_interface).)
\ No newline at end of file
+Every track at Exercism starts with ["Hello, World!"](http://en.wikipedia.org/wiki/%22Hello,_world!%22_program). Hello World is the traditional first program for beginning programmers. As programmers mature, they eventually want to test their code.
+
+Here at Exercism we encourage [Test Driven Development](http://en.wikipedia.org/wiki/Test-driven_development) (TDD). Test Driven Development is where you write your tests before writing any functionality. This allows you to consider the requirements of the code, and any edge cases that you may need to support.
+
+Once you finish your tests, you write the functionality. When all the functionality is finished, you can write tests for your next feature. One of the best features of tests is that you can very easily check for regression.
```
Comment (kytrinyx) [inline hello-world.md]: The opening of this paragraph kind of suggests that TDD is writing a bunch of tests up front, and then making all the tests pass afterwards.

Comment (kytrinyx) [inline hello-world.md]: It's totally clarified below, but it might be worth making it more explicit here as well.

File: hello-world.md [hunk 2]
```diff
@@ -1,15 +1,30 @@
...
+
+1. Run the test file and pick one test that's failing.
+2. Write some code to fix the test you picked.
+3. Re-run the tests to confirm the test is now passing.
+4. Repeat from step 1.
+5. [Submit your solution](http://help.exercism.io/submitting-exercises.html).
+
+## Instructions
+
+It's important to remember that the goal is to make code as expressive and readable as we can. Therefore, code submitted should be as general as possible and not just try to pass the tests, but also work for any input that matches the specification.
```
Comment (kytrinyx) [inline hello-world.md]: This is actually a fine line... we don't want to encourage speculative generality / premature generalization etc, but we also don't want people to submit a solution to `bob` that is a massive case statement for all the explicit string inputs in the tests.

It's kind of like _general_ but not _flexible_.

File: hello-world.md [hunk 3]
```diff
@@ -1,15 +1,30 @@
...
+
+## Specification
+
+The `Hello World!` program will greet me, the caller.
+
+If I tell the program my name is Tom, it will greet me by saying "Hello, Tom!".
```
Comment (kytrinyx) [inline hello-world.md]: Should we use "alice" and "bob" here, as a nod to classic crypto descriptions?

**Comment by ramonh** [comment]
I've now squashed it - Thanks for the help!

**Comment by kytrinyx** [comment]
Beautiful, thank you!

**Comment by ramonh** [comment]
My pleasure!
